src/Symfony/Component/Validator/Constraints/ConditionalValidator.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\LogicException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+final class ConditionalValidator extends ConstraintValidator
+{
+    private $expressionLanguage;
+
+    public function __construct(ExpressionLanguage $expressionLanguage = null)
+    {
+        $this->expressionLanguage = $expressionLanguage;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof Conditional) {
+            throw new UnexpectedTypeException($constraint, Conditional::class);
+        }
+
+        if (null === $value) {
+            return;
+        }
+
+        $context = $this->context;
+        $variables = $constraint->values;
+        $variables['value'] = $value;
+        $variables['this'] = $context->getObject();
+
+        if ($this->getExpressionLanguage()->evaluate($constraint->condition, $variables)) {
+            $context->getValidator()->inContext($context)
+                ->validate($value, $constraint->constraints);
+        }
+    }
+
+    private function getExpressionLanguage(): ExpressionLanguage
+    {
+        if (null === $this->expressionLanguage) {
+            if (!class_exists(ExpressionLanguage::class)) {
+                throw new LogicException('Unable to use expressions as the Symfony ExpressionLanguage component is not installed.');
```

Comment:
```suggestion
                throw new LogicException(sprintf('The "symfony/expression-language" component is required to use the "%s" constraint.', __CLASS__));
```
Maybe move this check to `Conditional`'s constructor like it's done for `Expression`?

Reply:
There is already the check in the constructor.

Reply:
Indeed. This makes this check unnecessary then, because you can't reach this line without an instance of `Conditional`, which would have already performed the same check via its constructor.

Reply:
Actually we can reach this line with an instance of a class that extends `Conditional` and with different constructor.
I get it won't almost never happen but it seems logic in the context of this validator class.
To be honest, I took this check from https://github.com/symfony/symfony/blob/e0cadfbc8a8a6908142321734adca34a52717117/src/Symfony/Component/Validator/Constraints/ExpressionValidator.php#L74

Reply:
Right. Maybe the constraint class could be made `final`?

Reply:
I don't think I should make it `final` because it will make the class non extendable for further implementation.